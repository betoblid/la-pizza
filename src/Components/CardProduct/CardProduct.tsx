import { Heart } from "lucide-react";
import { Products } from "../../@types";
import { FormatPrice } from "../../utils/FormatPrice";
import { useCart } from "../../Hooks/useCart";
import { useFavorit } from "../../Hooks/useFavorit";
import { memo, useState } from "react";

export const CardProduct = memo(({ products }: Products) => {

    const { description, name, price, thumb } = products;
    const {AddCard } = useCart()
    const { AddFavorit } = useFavorit()
    const [favorit, setfavorit] = useState(false)
    return (

        <div className=" rounded-xl relative ">
            <div className="flex items-center justify-between flex-col text-center max-w-64 w-full shadow-sm rounded-2xl" >

                <div className="flex items-center justify-around w-full bg-gradient-to-b from-transparent from-40% to-[#FEFBF4] to-40%">
                    <div></div>
                    <figure>
                        <img src={thumb} alt={name} title={`pizza de ${name}`} className="w-20" />
                    </figure>
                    <button 
                    className="border-2 border-red-500 rounded-md bg-red-200"
                    onClick={() => {AddFavorit({products}); setfavorit((favorit) => !favorit)}}>
                        {
                            favorit ? <Heart className="text-red-500 bg-red-600" /> : <Heart className="text-red-500" /> 
                        }
                    </button>
                </div>
                <div className="bg-[#FEFBF4]">
                    <h3 className="font-bold font-inter text-xl mt-3">
                        {name}
                    </h3>
                    <p className="font-normal text-sm font-inter my-3">
                        {description}
                    </p>
                    <span className="text-2xl font-bold font-inter">
                        {FormatPrice(price)}
                    </span>
                </div>
                <div className="bg-gradient-to-b from-[#FEFBF4] from-50% to-transparent to-50% w-full pt-5" >
                    <button 
                    className="bg-btn-orange uppercase text-sm font-semibold font-inter px-6 py-2  rounded-3xl"
                    onClick={() => AddCard({products})}>
                        Comprar Agora
                    </button>
                </div>
            </div>
        </div>
    )

})