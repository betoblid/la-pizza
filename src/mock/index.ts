export const prodctMock = {
    id: "mb1224453",
    thumb: "https://a.imagem.app/oNVBav.png",
    name: "La Cuma",
    description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
    price: 29.90
}