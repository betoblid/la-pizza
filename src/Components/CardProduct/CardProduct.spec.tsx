import { render, screen } from "@testing-library/react"
import { CardProduct } from "./CardProduct"
import { prodctMock } from "../../mock"

const prodctsMock = prodctMock
describe("<CardProduct />", () => {

    it("Verificar se o card está na tela", () => {

        render(<CardProduct products={prodctMock}/>)
        //verifdicar se existe no documento
        expect(screen.getByRole("heading", {name: /la Cuma/i})).toBeInTheDocument();
        //verificar se exite price do product
        expect(screen.getByText(/29.90/i)).toBeInTheDocument()
        //verificar se tem uma descrição
        expect(screen.getByText(/Lorem ipsum/i)).toBeInTheDocument()
    })
    it("verificar se imagem está na tela", () => {
        render(<CardProduct products={prodctMock}/>)
        //verificar se a imagem está visivel
        expect(screen.getByAltText("La Cuma")).toBeInTheDocument()
    })

    it("criar um snapshot", () => {
        const { container } =  render(<CardProduct products={prodctMock}/>)
    
        expect(container).toMatchSnapshot()
    })
})