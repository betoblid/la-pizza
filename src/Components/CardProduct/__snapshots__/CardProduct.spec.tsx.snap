// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardProduct /> criar um snapshot 1`] = `
<div>
  <div
    class=" rounded-xl relative "
  >
    <div
      class="flex items-center justify-between flex-col text-center w-64 shadow-sm rounded-2xl"
    >
      <div
        class="flex items-center justify-around w-full bg-gradient-to-b from-transparent from-40% to-[#FEFBF4] to-40%"
      >
        <div />
        <figure>
          <img
            alt="La Cuma"
            class="w-20"
            src="https://a.imagem.app/oNVBav.png"
            title="pizza de La Cuma"
          />
        </figure>
        <button
          class="border-2 border-red-500 rounded-md bg-red-200"
        >
          <svg
            class="lucide lucide-heart text-red-500"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
            />
          </svg>
        </button>
      </div>
      <div
        class="bg-[#FEFBF4]"
      >
        <h3
          class="font-bold font-inter text-xl mt-3"
        >
          La Cuma
        </h3>
        <p
          class="font-normal text-sm font-inter my-3"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </p>
        <span
          class="text-2xl font-bold font-inter"
        >
          R$Â 29,90
        </span>
      </div>
      <div
        class="bg-gradient-to-b from-[#FEFBF4] from-50% to-transparent to-50% w-full pt-5"
      >
        <button
          class="bg-btn-orange uppercase text-sm font-semibold font-inter px-6 py-2  rounded-3xl"
        >
          Comprar Agora
        </button>
      </div>
    </div>
  </div>
</div>
`;
